app/frontend/src/components/ImageInput.tsx

```diff
@@ -0,0 +1,72 @@
+import { CircularProgress } from "@material-ui/core";
+import makeStyles from "@material-ui/core/styles/makeStyles";
+import Alert from "components/Alert";
+import React from "react";
+import { useMutation } from "react-query";
+
+import { service } from "../service";
+
+const useStyles = makeStyles(() => ({
+  input: {
+    display: "none",
+  },
+  label: {
+    alignItems: "center",
+    display: "flex",
+    justifyContent: "center",
+    width: "fit-content",
+  },
+  loading: {
+    position: "absolute",
+  },
+}));
+
+export interface ImageInputValues {
+  file: File;
+  filename: string;
+  key: string;
+  thumbnail_url: string;
+  full_url: string;
+}
+
+export interface ImageInputProps {
+  id?: string;
+  name?: string;
+  onChange: (values: ImageInputValues) => void;
+  children: React.ReactNode;
+}
+
+export function ImageInput({ id, name, children, onChange }: ImageInputProps) {
+  const classes = useStyles();
+  const inputId = `${id}-file-input`;
+  const mutation = useMutation<ImageInputValues, Error, File>((file: File) =>
+    service.api.uploadFile(file)
+  );
+
+  return (
+    <>
+      <input
+        className={classes.input}
+        accept="image/*"
+        id={inputId}
+        name={name}
+        type="file"
+        onChange={async (event) => {
+          if (!event.target.files?.length) return;
+          const file = event.target.files[0];
+          const response = await mutation.mutateAsync(file);
+          onChange(response);
```

Comment:
Not sure auto submitting like this is good UX - what if the user selects the wrong image and want to correct it?

Reply:
They can choose a new image - it does auto-upload, but the upload will only be placed in the resource (in the case, profile) when the form as a whole is submitted.

Reply:
What I mean is, this auto-submits and auto-receives a key, but the key isn't put into the profile automatically, just into the form state.

Reply:
If you're content with that and you approve, feel free to merge this pr

Reply:
Ahh I see, sounds like more of a problem for Aapeli's wallet on S3 bucket cost than actual technical problem here for me here 😂

Reply:
Although storage (half) joke aside, it's still a bit shitty for people on mobile data for this to always upload?